warehouse/oidc/models.py

```diff
@@ -0,0 +1,144 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sentry_sdk
+
+from sqlalchemy import Column, ForeignKey, String, orm
+from sqlalchemy.dialects.postgresql import UUID
+
+from warehouse import db
+from warehouse.packaging.models import Project
+
+
+class OIDCProviderProjectAssociation(db.Model):
+    __tablename__ = "oidc_provider_project_association"
+
+    oidc_provider_id = Column(
+        UUID(as_uuid=True),
+        ForeignKey("oidc_providers.id"),
+        nullable=False,
+        primary_key=True,
+    )
+    project_id = Column(
+        UUID(as_uuid=True), ForeignKey("projects.id"), nullable=False, primary_key=True
+    )
```

Comment:
N.B.: Let me know if there's a better way to do this many-many mapping. This is the pattern I'm familiar with, but there are probably others.